juicy-html.html

```diff
@@ -38,194 +38,203 @@
+                var child;
+                while (childNo--) {
+                    // this.stampedNodes[childNo].remove();
+                    child = this.stampedNodes[childNo];
+                    if (child.parentNode) {
+                        child.parentNode.removeChild(child);
+                    }
                 }
+                // forget the removed nodes
+                this.stampedNodes = null;
             }
-        };
 
-        JuicyHTMLPrototype.attachModel = function (model, arrayOfElements) {
-            arrayOfElements || (arrayOfElements = this.stampedNodes);
-            if (model === null || !arrayOfElements) {
-                return;
+            attributeChangedCallback(name, oldVal, newVal) {
+                if (this.isAttached) {
+                    switch (name) {
+                        case "model":
+                            if (typeof newVal === "string") {
+                                newVal = newVal ? JSON.parse(newVal) : null;
+                            }
+                            this.model = newVal;
+                            break;
+                        case "content":
+                            warnAboutDeprecatedContentOn(this);
+                            forwardContentValue(this, newVal);
+                            break;
+                        case "href":
+                            // skip pending request
+                            this.skipStampingPendingFile();
+                            if (newVal !== null) {
+                                this.removeAttribute('html');
+                                this._loadExternalFile(newVal);
+                            } else {
+                                this.clear();
+                            }
+                            break;
+                        case "html":
+                            if (newVal !== null) {
+                                this.removeAttribute('href');
+                                this.reattachTemplate_(newVal);
+                            } else {
+                                this.clear();
+                            }
+                            break;
+                    }
+                }
             }
-            for (var childNo = 0; childNo < arrayOfElements.length; childNo++) {
-                arrayOfElements[childNo].model = model;
+
+            attachModel(model, arrayOfElements) {
+                arrayOfElements || (arrayOfElements = this.stampedNodes);
+                if (model === null || !arrayOfElements) {
+                    return;
+                }
+                for (var childNo = 0; childNo < arrayOfElements.length; childNo++) {
+                    arrayOfElements[childNo].model = model;
+                }
             }
         };
+        /**
+         * Reference to pending request
+         * @type {XMLHttpRequest}
+         */
+        JuicyHTMLElement.prototype.pending = null;
```

Comment:
Any reason not to put these in the constructor?

Reply:
Any reason to put it in the constructor? With the prototype, it's available even before you create an instance, so it's easier to extend it, makes API more discoverable.

Reply:
Plus it's easier to document it with JS doc ;)